src/Microsoft.Performance.SDK/Processing/CustomDataProcessorBaseWithSourceParser.cs

```diff
@@ -131,41 +131,103 @@ public void EnableCooker(ISourceDataCookerFactory sourceDataCookerFactory)
             }
         }
 
-        /// <summary>
-        /// Returns data from a source data cooker registered with <see cref="SourceProcessingSession"/>.
-        /// </summary>
-        /// <typeparam name="TOutput">Data output type</typeparam>
-        /// <param name="dataOutputPath">Path to the data output</param>
-        /// <returns>Data output of the specified type</returns>
+        /// <inheritdoc/>
+        /// <remarks>
+        ///     Returns data from a source data cooker registered with <see cref="SourceProcessingSession"/>.
+        /// </remarks>
+        /// <exception cref="ArgumentException">
+        ///     <paramref name="dataOutputPath"/> does not target the data processor.
+        /// </exception>
         public TOutput QueryOutput<TOutput>(DataOutputPath dataOutputPath)
```

Comment:
why do we even have methods for querying data cookers on a custom data processor? seems like we're blurring boundaries between object functionality

Reply:
This does blur things a bit, you're correct and we can chat offline. But I'm going to resolve this comment and it's not new functionality to this PR.